pkg/controller/plan/adapter/base/doc.go

```diff
@@ -78,6 +78,8 @@ type Builder interface {
 	GetPopulatorTaskName(pvc *core.PersistentVolumeClaim) (taskName string, err error)
 	// Convert the PVC to a given format
 	ConvertPVCs(pvcs []core.PersistentVolumeClaim) (ready bool, err error)
+	// Get the instancetype name
+	InstanceTypeName(vmRef ref.Ref) (name string, err error)
```

Comment:
IIUC you got this wrong. Instancetypes are about resource sizing, e.g. how many cores and how much memory your VM gets. Preferences are about runtime preferences, resp. hardware configuration a specific OS expects. Throughout this whole PR you use the term `InstanceType`, when you are actually refering to preferences.